algorithms/linfa-nn/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+use linfa::Float;
+use ndarray::{Array2, ArrayView1};
+
+pub mod kdtree;
+pub mod linear;
+
+pub(crate) type Point<'a, F> = ArrayView1<'a, F>;
+
+pub trait NearestNeighbour<F: Float> {
+    // Returns nearest in order. Might want wrap in result or return iterator
+    fn k_nearest<'b>(&self, point: Point<'b, F>, k: usize) -> Vec<Point<F>>;
```

Comment:
should probably fallible later on. But I like the abstraction of several algorithms.